force-app/main/default/classes/UTIL_DuplicateMgmt.cls

```diff
@@ -0,0 +1,157 @@
+    * Will throw error if more than 50 SObjects are passed in.
+    * @param List<sObject> list of Sobjects that duplicates will be located for
+    * @return Map<SObject, BDI_DuplicateService.MatchDetail[]>
+    */
+    public static Map<SObject, MatchDetail[]> findDuplicates(List<SObject> records) {
+        // Initiate return map
+        Map<SObject, MatchDetail[]> sObjToDupeResults = new Map<SObject, MatchDetail[]>();
+
+        Datacloud.FindDuplicatesResult[] results;
+
+        // execute duplicate matching
+        results = Datacloud.FindDuplicates.findDuplicates(records);
+        
+        // Loop through results and create MatchDetail records to store the information.
+        for (Integer i = 0; i < records.size(); i++) {
+            Datacloud.FindDuplicatesResult findDupeResult = results[i];
+            SObject recordToMatch = records[i];
+
+            // Placing key in map for record to match.  If no matches are found the value will stay null.
+            sObjToDupeResults.put(recordToMatch, null);
+
+            for (Datacloud.DuplicateResult dupeResult : findDupeResult.getDuplicateResults()) {      
+                for (Datacloud.MatchResult matchResult : dupeResult.getMatchResults()) {
+                    for (Datacloud.MatchRecord matchRecord : matchResult.getMatchRecords()) {
+                        storeMatchDetail(sObjToDupeResults, recordToMatch, dupeResult, matchResult, matchRecord);
+                    }
+                }
+            }                        
+        }
+
+        return sObjToDupeResults;
+    }
+
+    /**
+    * @description Returns map of duplicate matching details for the records passed in.  Breaks down any list
+    * of SObjects over 50 into chunks of 50 that can be processed without error.
+    * @param List<sObject> list of Sobjects that duplicates will be located for
+    * @return Map<String, BDI_DuplicateService.MatchDetail[]>
+    */
+    public Map<SObject, MatchDetail[]> findDuplicatesBulk(List<SObject> records) {
+        Map<SObject, MatchDetail[]> dupeResults = new Map<SObject, MatchDetail[]>();
+        Integer nextObjToProcess = 0;
+        Integer objsRemaining = records.size();
+
+        // Continue looping while there are still records remaining to process.
+        while (objsRemaining > 0) {
+            SObject[] sObjBatch = new SObject[]{};
+
+            // Only process up to 50 records at a time
+            Integer objsToProcess = Math.min(objsRemaining,50);
+
+            for (Integer i = 0; i < objsToProcess; i++) {
+    
+                sObjBatch.add(records[nextObjToProcess]);
+                nextObjToProcess++;
+            }
+
+            objsRemaining = objsRemaining - objsToProcess;
+            // Call duplicate management logic
+            Map<SObject, MatchDetail[]> tempDupeResults = findDuplicates(sObjBatch);
+            
+            // Put results in main return map
+            for (SObject sObj : tempDupeResults.keySet()) {
```

Comment:
Can we call the `putAll` operation on the main return map here? since we are not really doing any manipulation of the `tempDupeResults` map

Reply:
Yeah that should work just as well.